### PR TITLE
cleanup: Remove definitions and move them to its relative places acco…

### PR DESCRIPTION
**Pull request by harshavardhana**
- Move fs-definitions.go and break them into fs-datatypes.go, fs-bucket-acl.go
  and fs-utils.go
- Move api-definitions.go to api-response.go, where they should be.
- Move web-definitions to its related handlers.

### CODE REVIEW

**Comment by abperiasamy** [comment]
Update the copyright year only when you make copyright worthy changes to the source file. Please resubmit the patch.

**Comment by harshavardhana** [comment]
> Update the copyright year only when you make copyright worthy changes to the source file. Please resubmit the patch.

Okay.

**Comment by harshavardhana** [comment]
@abperiasamy - http://sourceforge.net/error-404.html?project=mingw-w64-dgn sourceforge download failed for windows.

**Comment by harshavardhana** [comment]
Sourceforge is up again. @abperiasamy - can you review now?

File: api-response.go [hunk 1]
```diff
@@ -17,16 +17,198 @@
...
+	XMLName  xml.Name `xml:"http://s3.amazonaws.com/doc/2006-03-01/ LocationConstraint" json:"-"`
+	Location string   `xml:",chardata"`
+}
+
+// AccessControlPolicyResponse - format for get bucket acl response.
+type AccessControlPolicyResponse struct {
+	XMLName xml.Name `xml:"http://s3.amazonaws.com/doc/2006-03-01/ AccessControlPolicy" json:"-"`
+
+	AccessControlList struct {
+		Grant []Grant
```
Comment (abperiasamy) [inline api-response.go]: Something like this?
Grants []Grant `xml:"Grant"`

File: api-response.go [hunk 2]
```diff
@@ -17,16 +17,198 @@
...
+	maxObjectList = 1000
 )
 
+// LocationResponse - format for location response.
+type LocationResponse struct {
+	XMLName  xml.Name `xml:"http://s3.amazonaws.com/doc/2006-03-01/ LocationConstraint" json:"-"`
+	Location string   `xml:",chardata"`
+}
+
+// AccessControlPolicyResponse - format for get bucket acl response.
```
Comment (abperiasamy) [inline api-response.go:39]: Move this below Grant declaration.

File: api-response.go [hunk 3]
```diff
@@ -17,16 +17,198 @@
...
+	// The class of storage used to store the object.
+	StorageClass string
+
+	PartNumberMarker     int
+	NextPartNumberMarker int
+	MaxParts             int
+	IsTruncated          bool
+
+	// List of parts.
+	Part []*Part
```
Comment (abperiasamy) [inline api-response.go]: Same here, Part to Parts?

File: api-response.go [hunk 4]
```diff
@@ -17,16 +17,198 @@
...
+	Bucket             string
+	KeyMarker          string
+	UploadIDMarker     string `xml:"UploadIdMarker"`
+	NextKeyMarker      string
+	NextUploadIDMarker string `xml:"NextUploadIdMarker"`
+	EncodingType       string
+	MaxUploads         int
+	IsTruncated        bool
+	Upload             []*Upload
+	Prefix             string
```
Comment (abperiasamy) [inline api-response.go:135]: Upload to Uploads?

File: api-response.go [hunk 5]
```diff
@@ -17,16 +17,198 @@
...
+	Delimiter          string
+	CommonPrefixes     []*CommonPrefix
+}
+
+// ListBucketsResponse - format for list buckets response
+type ListBucketsResponse struct {
+	XMLName xml.Name `xml:"http://s3.amazonaws.com/doc/2006-03-01/ ListAllMyBucketsResult" json:"-"`
+	// Container for one or more buckets.
+	Buckets struct {
+		Bucket []*Bucket
```
Comment (abperiasamy) [inline api-response.go]: Bucket to Buckets

File: api-response.go [hunk 6]
```diff
@@ -17,16 +17,198 @@
...
+// CommonPrefix container for prefix response in ListObjectsResponse
+type CommonPrefix struct {
+	Prefix string
+}
+
+// Bucket container for bucket metadata
+type Bucket struct {
+	Name         string
+	CreationDate string
+}
```
Comment (abperiasamy) [inline api-response.go:169]: Comment Date format

File: api-response.go [hunk 7]
```diff
@@ -17,16 +17,198 @@
...
+
+// Object container for object metadata
+type Object struct {
+	ETag         string
+	Key          string
+	LastModified string
+	Size         int64
```
Comment (abperiasamy) [inline api-response.go:176]: Comment Date format

File: pkg/fs/fs-backend-metadata.go [hunk 8]
```diff
@@ -1,5 +1,5 @@
 /*
- * Minio Cloud Storage, (C) 2015 Minio, Inc.
+ * Minio Cloud Storage, (C) 2015, 2016 Minio, Inc.
```
Comment (abperiasamy) [inline pkg/fs/fs-backend-metadata.go:2]: If there is no change made in this file, we dont update copyright year.

Comment (harshavardhana) [inline pkg/fs/fs-backend-metadata.go:2]: > If there is no change made in this file, we dont update copyright year.

The file was moved from config to backend-metadata so it is valid.

File: pkg/ioutils/sort.go [hunk 9]
```diff
@@ -14,4 +14,26 @@
  * limitations under the License.
  */
 
-package main
+package ioutils
+
+import "sort"
+
+// SortUnique sorts data which results in distinct elements in data in sorted order.
+func SortUnique(data sort.Interface) {
```
Comment (abperiasamy) [inline pkg/ioutils/sort.go]: SortUnique is supposed to remove duplicate entries. This function is swapping already sorted entries. What is the outcome?

File: pkg/ioutils/ioutils_test.go [hunk 10]
```diff
@@ -32,6 +33,14 @@ type MySuite struct{}
 
 var _ = Suite(&MySuite{})
 
+func (s *MySuite) TestSortUnique(c *C) {
+	intSlice := []int{1, 1, 2, 3, 4, 5, 5, 6, 7, 8, 9, 9, 10, 10}
```
Comment (abperiasamy) [inline pkg/ioutils/ioutils_test.go]: []int{1, 1, 2, 3, 4, 5, 5, 5, 5, 5, 6, 7, 8, 9, 9, 10, 10}

Comment (harshavardhana) [inline pkg/ioutils/ioutils_test.go]: > []int{1, 1, 2, 3, 4, 5, 5, 5, 5, 5, 6, 7, 8, 9, 9, 10, 10}

Even this works.

File: api-response.go [hunk 11]
```diff
@@ -17,16 +17,198 @@
...
+	"encoding/xml"
 	"net/http"
 
 	"github.com/minio/minio/pkg/fs"
 )
 
-// Reply date format
 const (
+	// Reply date format
 	rfcFormat = "2006-01-02T15:04:05.000Z"
```
Comment (abperiasamy) [inline api-response.go]: Is this RFC3339 format? In that case,  variable should also mention the rfc name.

Comment (harshavardhana) [inline api-response.go]: No AB it is not fully rfc3339. S3 date doesn't have nanosecond element. It's a shorter version of rfc3339 perhaps.

**Pull request merged**